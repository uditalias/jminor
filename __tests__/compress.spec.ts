'use strict';

import createNumericKeyGenerator from "../lib/generators/numericKeyGenerator";
import { createDictionary, compress } from "../lib";

const data = require("./__mocks__/data.json");

describe('compress', () => {

    let dictionary;
    beforeEach(() => {
        dictionary = createDictionary().fromJSON(data);
    });

    it('should compress data', () => {

        const compressed = compress(data, dictionary, {});

        expect(compressed).toMatchSnapshot();

    });

    it('should compress data without false values', () => {

        const compressed = compress(data, dictionary, {
            boolean: {
                removeFalse: true
            }
        });

        expect(compressed).toMatchSnapshot();

    });

    it('should compress data without empty strings', () => {

        const compressed = compress(data, dictionary, {
            string: {
                removeEmpty: true
            }
        });

        expect(compressed).toMatchSnapshot();

    });

    it('should compress data without zero values', () => {

        const compressed = compress(data, dictionary, {
            number: {
                removeZero: true
            }
        });

        expect(compressed).toMatchSnapshot();

    });

    it('should compress data without null values', () => {

        const compressed = compress(data, dictionary, {
            null: {
                removeNull: true
            }
        });

        expect(compressed).toMatchSnapshot();

    });

    it('should not compress data', () => {

        const compressed = compress(data, dictionary, {
            translateKeys: false
        });

        expect(compressed).toMatchSnapshot();

    });

    it('should compress with numeric key generator', () => {

        dictionary = createDictionary(createNumericKeyGenerator).fromJSON(data);;

        const compressed = compress(data, dictionary);

        expect(compressed).toMatchSnapshot();
    });
});