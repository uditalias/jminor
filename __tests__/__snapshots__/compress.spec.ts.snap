// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compress should compress and remove empty arrays 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
  },
  "k": Object {
    "l": false,
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress and remove empty objects 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "j": Array [],
  },
  "k": Object {
    "l": false,
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress and remove empty result array 1`] = `
Object {
  "a": 1,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
  },
  "k": Object {
    "l": false,
  },
}
`;

exports[`compress should compress and remove empty result object 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "j": Array [],
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress data 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
    "j": Array [],
  },
  "k": Object {
    "l": false,
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress data without empty strings 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
    "j": Array [],
  },
  "k": Object {
    "l": false,
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress data without false values 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
    "j": Array [],
  },
  "k": Object {},
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress data without null values 1`] = `
Object {
  "a": 1,
  "b": 0,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "g": Object {
    "h": "test",
    "i": Object {},
    "j": Array [],
  },
  "k": Object {
    "l": false,
  },
  "m": Array [
    Object {
      "n": 0,
    },
  ],
}
`;

exports[`compress should compress data without zero values 1`] = `
Object {
  "a": 1,
  "c": Array [
    Object {
      "d": 4,
      "e": "",
    },
    Object {
      "d": true,
      "e": false,
    },
  ],
  "f": null,
  "g": Object {
    "h": "test",
    "i": Object {},
    "j": Array [],
  },
  "k": Object {
    "l": false,
  },
  "m": Array [],
}
`;

exports[`compress should compress with numeric key generator 1`] = `
Object {
  "1": 1,
  "11": Object {
    "12": false,
  },
  "13": Array [
    Object {
      "14": 0,
    },
  ],
  "2": 0,
  "3": Array [
    Object {
      "4": 4,
      "5": "",
    },
    Object {
      "4": true,
      "5": false,
    },
  ],
  "6": null,
  "7": Object {
    "10": Array [],
    "8": "test",
    "9": Object {},
  },
}
`;

exports[`compress should not compress data 1`] = `
Object {
  "first": 1,
  "fourteen": Array [
    Object {
      "fifteen": 0,
    },
  ],
  "second": 0,
  "seventh": Object {
    "eighth": "test",
    "eleventh": Array [],
    "tenth": Object {},
  },
  "sixth": null,
  "third": Array [
    Object {
      "fifth": "",
      "fourth": 4,
    },
    Object {
      "fifth": false,
      "fourth": true,
    },
  ],
  "twelve": Object {
    "thirteen": false,
  },
}
`;
